server/src/main/java/io/crate/metadata/IndexReference.java

```diff
@@ -210,6 +216,11 @@ public Map<String, Object> toMapping() {
             mapping.put("analyzer", analyzer);
             mapping.put("type", "text");
         }
+
+        if (columns.isEmpty() == false) {
+            mapping.put("sources", columns.stream().map(ref -> ref.column().fqn()).collect(Collectors.toList()));
+        }
+
```

Comment:
On the new `CreateTableRequest` creation we will be able to set `columns` accordingly